RFCs/0010-Source-Information-For-Discovered-Tests.md

```diff
@@ -0,0 +1,71 @@
+Test adapters can rely on this flag and skip source information collection for discovered tests, if the flag is set to 'false'. In such cases, Visual Studio Test Explorer
+will determine source information for the discovered tests.
+
+In case of CLI, source information is not required. Hence this flag is set to 'false' in the runsettings.
+
+### A sample runsettings file with this flag:
+```xml
+<RunSettings>
+    <RunConfiguration>
+      <CollectSourceInformation>False</CollectSourceInformation>
+    </RunConfiguration>
+</RunSettings>
+```
+
+Following are the perf numbers obtained for this flag in IDE scenario for MSTest and XUnit test adapters for a test project having 25K tests:
+
+| Adapter         | Test Platform | Without Changes = A (Sec) | With Changes = B (Sec) | Improvement, C = A - B (Sec) | Improvement (%) = C * 100 / A |
+|-----------------|---------------|---------------------------|------------------------|------------------------------|-------------------------------|
+| MSTest V2       | TPv1          |   7.72                    |   6.20                 |  1.02                        | 14.12                         |
+| MSTest V2       | TPv2          |  27.28                    |  26.09                 |  1.19                        |  4.36                         |
+| XUnit           | TPv1          | 252.96                    | 195.30                 | 57.66                        | 22.79                         |
+| XUnit           | TPv2          | 129.97                    |  79.32                 | 50.65                        | 38.97                         |
+
+
+## How to test this change
+While the changes for CLI will be available in a future release, changes for IDE are already available in Visual Studio 2017 Update 3 Preview 3 and newer builds but is hidden behind a feature flag.
+Steps to turn ON the feature flag:
+1. Close all instances of Visual Studio
+2. Locate Microsoft.VisualStudio.FeatureFlags.pkgdef file in Visual Studio installation folder. This file generally gets installed at:
+C:\Program Files (x86)\Microsoft Visual Studio\Preview\Enterprise\Common7\IDE\CommonExtensions\Platform\Shell
+3. Take a backup of this file
+4. Open the file in a text editor and location following key:
+[$RootKey$\FeatureFlags\TestingTools\UnitTesting\UseSipToFetchSourceInformation]
+5. Update its dword value from 00000000 to 00000001
+6. Save the file and update it in the Visual Studio location mentioned in (2)
+7. Open Developer Command prompt in administrator mode and run following command:
+   devenv /UpdateConfiguration
+8. Launch Visual Studio and do test discovery for a managed (C# or VB) test project
+
+To disable the feature, restore backup file in the Visual Studio location mentioned in (2) and run step (7)
+
+Note that for perf improvement to show up, the test adapter needs to be updated to honor 'CollectSourceInformation' flag.
```

Comment:
It will be good to mention the adapter version here which supports this flag i.e. 1.2.0-beta onwards.